packages/charts/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx

```diff
@@ -780,11 +760,61 @@ export class HorizontalBarChartWithAxisBase extends React.Component<
         focusZonePropsInHoverCard={this.props.focusZonePropsForLegendsInHoverCard}
         overflowText={this.props.legendsOverflowText}
         {...this.props.legendProps}
+        canSelectMultipleLegends={this.props.legendProps?.canSelectMultipleLegends}
+        onChange={this._onLegendChange}
       />
     );
     return legends;
   };
 
+  private _isLegendSelected = (): boolean => {
+    return this.state.isLegendSelected!;
+  };
+
+  private _isSpecificLegendTitleSelected = (legend: string): boolean => {
+    return this.state.selectedLegendTitle === legend;
+  };
+
+  private _isLegendHovered = (): boolean => {
+    return this.state.isLegendHovered!;
+  };
+
+  private _isSpecificLegendSelected = (legend: string): boolean => {
+    return this.state.selectedLegends.indexOf(legend) > -1;
+  };
+
+  private _noLegendsSelected = (): boolean => {
+    return this.state.selectedLegends.length === 0;
+  };
+
+  private _onLegendChange = (
+    selectedLegends: string[],
+    event: React.MouseEvent<HTMLButtonElement>,
+    currentLegend?: ILegend,
+  ) => {
+    this.setState({ selectedLegends });
+    if (this._isLegendSelected()) {
+      if (this._isSpecificLegendTitleSelected(currentLegend?.title!)) {
+        this.setState({
+          isLegendSelected: false,
```

Comment:
>isLegendSelected: false [](http://example.com/codeflow?start=10&length=23)

why setting this to false if setting selectedLegendTitle to a value

Reply:
updated